sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala

```diff
@@ -89,8 +90,11 @@ class UnivocityGenerator(
     case udt: UserDefinedType[_] => makeConverter(udt.sqlType)
 
     case dt: DataType =>
-      (row: InternalRow, ordinal: Int) =>
-        row.get(ordinal, dt).toString
+      (row: InternalRow, ordinal: Int) => {
+        val valueUTF8String = ToPrettyString(Literal(row.get(ordinal, dt), dt),
```

Comment:
Does `ToPrettyString` prints nested fields in the same way as `UnivocityGenerator`. For instance, does it follow to datetime patterns specified for the CSV datasource?

Reply:
Let me think again, maybe we need to rewrite the logic similar to `ToPrettyString ` in `UnivocityGenerator ` based on the characteristics of CSV.

Reply:
I changed the approach and enabled `UnivocityGenerator ` to support complex type (`ArrayType`, `MapType`, `StructType`) outputs.